### PR TITLE
Add debug mode benchmark

### PR DESCRIPTION
**Pull request by ksztenderski**
Signed-off-by: ksztenderski <ksztenderski@nvidia.com>

## Category:

**Other** (*e.g. Documentation, Tests, Configuration*)

## Description:

Adds benchmark comparing standard and debug mode.

I measured performance of standard and debug modes on 2 pipelines defined in the benchmark.
Measurement parameters:

- batch_sizes - [1, 4, 8, 32, 64, 128]
- num_threads - [1, 2, 4, 8]
- num_samples - 2048
- dataset - `DALI_extra/db/single/jpeg`

Number of iterations for each pipeline is defined as `num_samples/batch_size`.

Legend:
- standard - standard pipeline with `exec_pipelined=True` and `exec_async=True`
- standard_sync - standard pipeline with `exec_pipelined=False` and `exec_async=False`
- debug - new debug pipeline using eager operators
- debug_old - old debug pipeline using minipipelines for each operator

### 1 thread
avg_speed_threads_1
iter_time_threads_1
time_threads_1

### 2 threads
avg_speed_threads_2
iter_time_threads_2
time_threads_2

### 4 threads
avg_speed_threads_4
iter_time_threads_4
time_threads_4

### 8 threads
avg_speed_threads_8
iter_time_threads_8
time_threads_8

### 
2_iter_time_threads_1
2_time_threads_1
2_avg_speed_threads_1
1 thread

### 2 threads
2_avg_speed_threads_2
2_iter_time_threads_2
2_time_threads_2

### 4 threads
2_avg_speed_threads_4
2_iter_time_threads_4
2_time_threads_4

### 8 threads
2_avg_speed_threads_8
2_iter_time_threads_8
2_time_threads_8

### Tests
- [ ] Existing tests apply
- [x] New tests added
  - [x] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [x] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

**Comment by ksztenderski** [comment]
!build

File: dali/test/python/test_pipeline_debug_resnet50.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    times[1] = time()
+
+    for i in range(num_iters):
+        pipe.run()
+        times[i + 2] = time()
+
+    full_time = times[-1] - times[1]
+    times = np.diff(times)
+
+    return full_time, times[0 + debug], times[1:]
```
Comment (stiepan) [inline dali/test/python/test_pipeline_debug_resnet50.py:90]: So we treat the time of a first run as a build time of the debug pipline, right?

File: dali/test/python/test_pipeline_debug_resnet50.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+from test_utils import get_dali_extra_path
+
+
+@pipeline_def(device_id=0)
+def rn50_pipeline(data_path):
+    uniform = fn.random.uniform(range=(0., 1.), shape=2)
+    resize_uniform = fn.random.uniform(range=(256., 480.))
+    mirror = fn.random.coin_flip(probability=0.5)
+    jpegs, _ = fn.readers.file(
+        file_root=data_path, shard_id=0, num_shards=2)
```
[the remark this answers is not in the document]
Comment (ksztenderski) [reply dali/test/python/test_pipeline_debug_resnet50.py:34]: No reason other than I had a test pipeline with file reader initialized this way. Changed it to `fn.readers.file(file_root=data_path)`.

**Comment by ksztenderski** [comment]
!build

File: dali/test/python/test_pipeline_debug_resnet50.py (hunk 1 above)
Comment (ksztenderski) [reply dali/test/python/test_pipeline_debug_resnet50.py]: done

**Comment by ksztenderski** [comment]
!build

**Comment by mzient** [comment]
I know it's quite masochistic, but I'd very much like to see the performance for even smaller batches (1, 4). Batch of 1 _is_ going to be used a lot with debug mode, we can be pretty sure of that.

**Review by awolant — APPROVED** [review]

**Review by stiepan — APPROVED** [review]

**Comment by ksztenderski** [comment]
!build

**Pull request merged**